acceptance/tests/utf8/utf8-in-file-resource.rb

```diff
@@ -0,0 +1,55 @@
+test_name 'utf-8 characters in resource title and param values' do
+  confine :except, :platform => [ 'windows', 'ubuntu-16']   # PUP-6983
+  utf8chars = "€‰ㄘ万竹ÜÖ"
+  agents.each do |agent|
+    agent_file = agent.tmpfile("file" + utf8chars) 
+    # remove this file, so puppet can create it and not merely correct
+    # its drift.
+    on(agent, "rm -rf #{agent_file}")
+
+    manifest =
+<<PP
+
+file { "#{agent_file}" :
+  ensure => file,
+  mode => "0644",
+  content => "This is the file content. file #{utf8chars} 
```

Comment:
In the context of the comment above about using "\u" notation, to ensure the interpolation happens we can also use heredoc with \u escape sequences enabled, i.e., 

https://docs.puppet.com/puppet/latest/lang_data_string.html#enabling-escape-sequences

If that, this would be something like:
```ruby
$file_content = @("doc"/$u)
This is the utf8 file content: #{utf8chars}
| doc

...
mode => "0644",
content => $file_content, 
}
```